src/js/components/pipelines/FacetDropzone.tsx

```diff
@@ -0,0 +1,76 @@
+import * as React from 'react';
+import { connect } from 'react-redux';
+import {State} from '../../store';
+import {FieldDraggingStateRecord} from '../../store/factory/Inspector';
+// import {GroupFacet} from "../../store/factory/marks/Group";
+import {Facet} from 'vega-typings';
+import {getClosestGroupId} from '../../util/hierarchy';
+import {addFacetLayout} from '../../actions/facetLayoutActions';
+import {addFacet} from '../../actions/markActions';
+import {FacetLayout} from '../../store/factory/FacetLayout';
+interface StateProps {
+  dragging: FieldDraggingStateRecord;
+  groupId: number;
+}
+
+interface OwnProps {
+  layoutOrientation: string
+}
+interface DispatchProps {
+  facetField: (field: string, groupId: number) => void;
+}
+
+function mapStateToProps(state: State): StateProps {
+  const groupId = getClosestGroupId();
+
+  const draggingRecord = state.getIn(['inspector', 'dragging']);
+  const isFieldDrag = draggingRecord && (draggingRecord as FieldDraggingStateRecord).dsId;
+
+  return {
+    dragging: isFieldDrag ? draggingRecord : null,
+    groupId
+  };
+}
+
+function mapDispatchToProps(dispatch, ownProps: OwnProps): DispatchProps {
+  return {
+    facetField: (field, groupId) => {
+      let numCols;
+      if (ownProps.layoutOrientation == "Column") {
+        numCols = 1;
+      } else {
+        numCols = null;
+      }
+      dispatch(addFacetLayout(FacetLayout({columns: numCols})));
+      // dispatch(addGroupFacet(GroupFacet({facet: {name: "facet", data: "cars_source_5", groupby: [field]}}), groupId)); // remove hardcoded data name
+      dispatch(addFacet({name: "facet",data: "5", groupby: field} as Facet, groupId));
```

Comment:
How to lookup lyra ID for dataset associated with dropped field?

Reply:
see comment in `handleDrop` below